Core Team Notes 2025-01-10 (#4301)
tock/tock: doc/wg/core/notes/core-notes-2025-01-10.md

# Tock Meeting Notes 2025-01-10

## Attendees

- Johnathan Van Why
- Amit Levy
- Leon Schuermann
- Tyler Potyondy
- Branden Ghena
- Alexandru Radovici
- Benjamin Prevor
- Brad Campbell
- Hudson Ayers

## Updates

### Tock Release 2.2:
- Leon: Tock 2.2 is released. 2 years between Tock 2.1 and Tock 2.2. Please let me know if there are any issues people find with this release.

### Treadmill:

> - Leon: Treadmill is now generating reports for PRs. We now have a 3rd github action that reports success and does nothing else. This 3rd github action is used for testing. This is a bit hacky, but is mostly working.

Reviewer: ```suggestion
- Leon: Treadmill is now a required check for merging PRs. We now have a 3rd github action that reports success and does nothing else for pushes to a PR, and performs actual tests when merging. This is a bit hacky, but is mostly working.
```

- Branden: Now Treadmill doesn't run until someone clicks merge?
- Leon: It never ran before merge, we cannot have it run automatically. We have a planned change to test PRs when they are opened.
- Branden: Is there a way for developers to manually run tests on branches?
- Leon: This is only possible for branches that originate from the Tock repository.
- Leon: Failures on PRs when merging are now reported and placed within the PR.
- Leon: Please let Ben or myself know if anyone experiences issues or failures with Treadmill.

### PSOC Board Support (https://github.com/tock/tock/pull/4300):
- Alex: We added an infineon board. This is the PSOC. 

> - Leon: I saw this has a cortex-m0+ and a cortex-m4. Is there any reason you targeted this?

Reviewer: ```suggestion
- Leon: I saw this has a cortex-m0+ and a cortex-m4. Is there any reason you targeted the M0+?
```

- Alex: I will get back to you on this. I think it has something to do with how the code is uploaded.
- Branden: I remember PSCOC having some hardware state machine similar to the RP2040. Are you going to try to support this?
- Alex: No, not at this time. We selected this because we wanted to support an infineon chip and this was the chip infineon recommended us to work with. 

### Meeting Logistics:
- Amit: Moving forward we will switch these meetings to be Wednesdays, potentially moving the time 15-30 min later.

> - Amit: This change will start next week Wednesday. I will send an announcement.

Reviewer 2: Should probably include that we moved the meeting to start at 9:15am Pacific Time (15 minutes later than the previous time) and shortened it to 45 minutes so it still ends at 10am.

- Leon: As we add more boards and tests, I don't think we can or should run the full suite of all tests for every PR as this would take too long.
- Leon: We need to develop an intelligent way for determining how to select which tests need to be executed for a given PR.
- Amit: In terms of coverage, how many boards is treadmill testing?
- Leon: We have a wip imix and litex boards. NRF is the only automated board.
- Leon: Many of these tests are UART and GPIO. 
- Leon: Other tests require uncommenting lines in the kernel. We haven't tackled this problem.
- Branden: One of the original goals was parallelism. If we had multiple NRF boards or boards at different universities, what is our status with this?
- Leon: We can add this using github actions to have a python script that uses some plan / config to match tests to available board resources.
- Branden: Having this would help with the testing bottle neck and also improve the stability of the system.
- Tyler: Would Treadmill's design cause headaches when deploying across universities (such as with campus IT security concerns).

> - Leon: All outgoing traffic goes through a VPN. One way we can mitigate concerns from campus IT security is by only allowing Tock CI testing jobs to run. This would mean we are not giving users access to execute arbitrary code 

Reviewer: ```suggestion
- Leon: All outgoing traffic goes through a VPN. One way we can mitigate concerns from campus IT security is by only allowing Tock CI testing jobs to run. This would mean we are not giving users direct access to execute arbitrary code 
```

This is a bit subtle, our CI workflows are still just arbitrary code being executed (hence all of our efforts to have Treadmill provide ephemeral environment where it's hard to persist changes across jobs)
